include/bluetooth/audio/csis.h

```diff
@@ -193,121 +193,135 @@ int bt_csis_advertise(struct bt_csis *csis, bool enable);
+	 *
+	 * The Set SIRK may not be exposed by the server over Bluetooth, and
+	 * may require an out-of-band solution.
+	 */
+	uint8_t set_sirk[BT_CSIS_SET_SIRK_SIZE];
+
+	/**
+	 * @brief The size of the set
+	 *
+	 * Will be 0 if not exposed by the server.
+	 */
 	uint8_t set_size;
+
+	/**
+	 * @brief The rank of the set on on the remote device
+	 *
+	 * Will be 0 if not exposed by the server.
+	 */
 	uint8_t rank;
+
+	/** Internally used pointer value */
+	struct bt_csis *csis;
 };
 
+/** Struct representing a remote device as a set member */
 struct bt_csis_client_set_member {
+	/** Connection pointer to the remote device, populated by the user */
 	struct bt_conn *conn;
-	bt_addr_le_t addr;
+	/** Array of sets for the remote device */
 	struct bt_csis_client_set sets[BT_CSIS_CLIENT_MAX_CSIS_INSTANCES];
 };
 
-typedef void (*bt_csis_client_discover_cb)(struct bt_conn *conn, int err,
-					   uint8_t set_count);
+/**
+ * @typedef bt_csis_client_discover_cb
+ * @brief Callback for discovering Coordinated Set Identification Services.
+ *
+ * @param member    Pointer to the set member.
+ * @param err       0 on success, or an errno value on error.
+ * @param set_count Number of sets on the member.
+ */
+typedef void (*bt_csis_client_discover_cb)(struct bt_csis_client_set_member *member,
+					   int err, uint8_t set_count);
 
 /**
  * @brief Initialise the csis_client instance for a connection. This will do a
  * discovery on the device and prepare the instance for following commands.
  *
- * @param member Pointer to a set member struct to store discovery results in
+ * @param member Pointer to a set member struct to store discovery results in.
  *
  * @return int Return 0 on success, or an errno value on error.
  */
 int bt_csis_client_discover(struct bt_csis_client_set_member *member);
 
-typedef void (*bt_csis_client_discover_sets_cb)(struct bt_conn *conn,
-						int err, uint8_t set_count,
-						struct bt_csis_client_set *sets);
+/**
+ * @typedef bt_csis_client_discover_sets_cb
+ * @brief Callback for discovering sets and their content.
+ *
+ * @param member    Pointer to the set member.
+ * @param err       0 on success, or an errno value on error.
+ * @param set_count Number of sets on the member.
+ */
+typedef void (*bt_csis_client_discover_sets_cb)(struct bt_csis_client_set_member *member,
+						int err, uint8_t set_count);
 
 /**
  * @brief Reads CSIS characteristics from a device, to find more information
  * about the set(s) that the device is part of.
  *
- * @param conn The connection to the device to read CSIS characteristics
+ * @param member Pointer to a set member struct to store discovery results in.
  *
  * @return int Return 0 on success, or an errno value on error.
  */
-int bt_csis_client_discover_sets(struct bt_conn *conn);
-
-typedef void (*bt_csis_client_discover_members_cb)(int err, uint8_t set_size,
-						   uint8_t members_found);
+int bt_csis_client_discover_sets(struct bt_csis_client_set_member *member);
```

Comment:
The difference between `discover()` and `discover_sets()` is not fully clear to me.
I believe the first one is to be called first, but they take the same input, and their callbacks return the same values.

Reply:
Agreed, they are very similar, and actually both of them are needed for the discovery procedure. 

The first, `discover()` discover the CSIS instances, if any, and initializes the handles. 
The second `discover_sets()` continues the discovery procedure by reading the SIRK, size and rank (if exposed). 

We could consider merging these two, as they are the same procedure. It was intentionally split into two in case that the device did not want to read all the characteristic during discovery, but maybe we need to revisit that idea. Merging the two may be needed to be fully compliant, as a set coordinator _shall_ read the characteristics during the discovery procedure. 

So the first discovers the CSIS service and the GATT handles, the second continues the discovery procedure which includes reading the characteristics. 

**So should we merge the two into a single function (which is more complex), or just rename this one into something better (suggestions are welcome)?**

Reply:
I vote to keep them separate, but update name and documentation.
Proposal:
"discover()" -> "discover_service()", with the documentation saying that
- this is the first discovery function to be run, and that it discovers the existence of the service, but not the characteristics of the service
"discover_sets()" keeps its name, documentation extended to say that it is run after "discover_service()", and discover sets by discovering SIRK, size and rank.

Reply:
> "discover()" -> "discover_service()", with the documentation saying that

I am strongly against this one, unless we update it for the other clients as well. 

How about we rename `discover_sets` to `read_sets` which is more aligned to what it actually does (read the set characteristics). 

If we keep them separate, `discover()` will not be compliant with the discover procedure (Coordinated Set Discovery procedure) which is a good reason to merge them.

Reply:
> > "discover()" -> "discover_service()", with the documentation saying that
> 
> I am strongly against this one, unless we update it for the other clients as well.

Consistency - good point.  I agree.

> How about we rename `discover_sets` to `read_sets` which is more aligned to what it actually does (read the set characteristics).

Seems fine.  I still think the documentation of both could be updated with a little bit of your explanation above.
> 
> If we keep them separate, `discover()` will not be compliant with the discover procedure (Coordinated Set Discovery procedure) which is a good reason to merge them.

You have convinced me.  Having a single function to do the procedure seems cleaner.  And I do not really see now why one would want to find the services, but not their content.  So merge them.

Reply:
> And I do not really see now why one would want to find the services, but not their content.

That is more typical for the rest of the LE Audio services though. Sometimes you want to discover the service, and only read specific values during discovery (unlike CSIP which requests to read _all_). 

I'll merge the two functions and update this PR.